src/modules/cairo/generator.js

```diff
@@ -415,7 +415,7 @@ function getInArgumentSource(p, n) {
     return `auto ${p.name} = Nan::To<int64_t>(info[${n}].As<Number>()).ToChecked();`
 
   if (typeName === 'const char *')
-    return `auto ${p.name} = *Nan::Utf8String (info[${n}].As<String>());`
+    return `auto ${p.name}__value = info[${n}].As<String>(); auto ${p.name} = *Nan::Utf8String(${p.name}__value);`
```

Comment:
It is still not working: The `Nan::Utf8String` has no name and gets freed too soon.

It should be something like this:
```js
return `Nan::Utf8String ${p.name}__value(info[${n}].As<String>()); auto ${p.name} = *${p.name}__value;`
```

(I haven't tried the generation script, yet. But that's what my modification in the code would look like.)

Another small thing: Would it be possible to add a newline (and proper indentation) there?

Reply:
Updated.

For the indentation, it would require more work that I can do at the moment. Some parts of the generator are able to return array of lines that get indented correctly, but overall indentation is messy and would need more work.